static/app/components/profiling/profilingTransactionHovercard.tsx

```diff
@@ -0,0 +1,301 @@
+import Link from '../links/link';
+import LoadingIndicator from '../loadingIndicator';
+import PerformanceDuration from '../performanceDuration';
+
+interface ProfilingTransactionHovercardProps {
+  organization: Organization;
+  project: Project;
+  transaction: string;
+}
+
+export function ProfilingTransactionHovercard(props: ProfilingTransactionHovercardProps) {
+  const {project, transaction, organization} = props;
+  const {query} = useLocation();
+
+  const linkToSummary = generateProfileSummaryRouteWithQuery({
+    query,
+    orgSlug: organization.slug,
+    projectSlug: project.slug,
+    transaction,
+  });
+
+  const triggerLink = <Link to={linkToSummary}>{transaction}</Link>;
+
+  if (!organization.features.includes('profiling-dashboard-redesign')) {
+    return triggerLink;
+  }
+
+  return (
+    <StyledHovercard
+      delay={250}
+      header={
+        <Flex justify="space-between" align="center">
+          <TextTruncate title={transaction}>{transaction}</TextTruncate>
+          <Button to={linkToSummary} size="xs">
+            {t('View Profiles')}
+          </Button>
+        </Flex>
+      }
+      body={
+        <ProfilingTransactionHovercardBody
+          transaction={transaction}
+          project={project}
+          organization={organization}
+        />
+      }
+      showUnderline
+    >
+      {triggerLink}
+    </StyledHovercard>
+  );
+}
+
+function ProfilingTransactionHovercardBody({
+  transaction,
+  project,
+  organization,
+}: ProfilingTransactionHovercardProps) {
+  const {selection} = usePageFilters();
+
+  const baseQueryOptions = {
+    query: `transaction:"${transaction}"`,
+    fields: getProfilesTableFields(project.platform),
+    enabled: Boolean(transaction),
+    limit: 1,
+    referrer: 'api.profiling.profiling-transaction-hovercard',
+    refetchOnMount: false,
+    projects: [project.id],
+  };
+
+  const slowestProfileQuery = useProfileEvents({
+    ...baseQueryOptions,
+    sort: {
+      key: 'profile.duration',
+      order: 'desc',
+    },
+  });
+
+  const latestProfileQuery = useProfileEvents({
+    ...baseQueryOptions,
+    sort: {
+      key: 'timestamp',
+      order: 'desc',
+    },
+  });
+
+  const functions = useFunctions({
+    project,
+    query: '',
+    selection,
+    transaction,
+    sort: '-p99',
+    functionType: 'application',
+  });
+
+  const slowestProfile = slowestProfileQuery?.data?.[0].data[0] ?? null;
+  const durationUnits = slowestProfileQuery.data?.[0].meta.units['profile.duration'];
+  const multiplier = durationUnits ? DURATION_UNITS[durationUnits] ?? 1 : 1;
+
+  const latestProfile = latestProfileQuery?.data?.[0].data[0] ?? null;
+
+  const linkToFlamechartRoute = (
+    profileId: string,
+    query?: {frameName: string; framePackage: string}
+  ) => {
+    return generateProfileFlamechartRouteWithQuery({
+      orgSlug: organization.slug,
+      projectSlug: project.slug,
+      profileId,
+      query,
+    });
+  };
```

Comment:
maybe its time for a `useProfilingRoutes` that handles `orgSlug` and `projectSlug` for us.